### PR TITLE
Nova chance: Ensure that there is always a currentUser object

### PR DESCRIPTION
**Pull request by GregWeil**
If currentUser would be null, instead act like we're an anon user. The two differences between this and a real anon is the id (`0`) and the avatar color (`#aaa`). The user pop is visible while signed out now, but it is disabled.

There's probably a lot of potential cleanup, I just hit the main 'do not render if there is no user' spots.

### CODE REVIEW

File: src/presenters/current-user.jsx [hunk 1]
```diff
@@ -181,10 +192,9 @@ class CurrentUserManager extends React.Component {
   
   render() {
     const {children, sharedUser, cachedUser, setSharedUser, setCachedUser} = this.props;
-    const currentUser = cachedUser || sharedUser;
     return children({
       api: this.api(),
-      currentUser: currentUser ? cleanUser(currentUser) : null,
+      currentUser: {...defaultUser, ...sharedUser, ...cachedUser},
```
Comment (judeallred) [inline src/presenters/current-user.jsx]: ahahahah, fun!

File: src/presenters/pop-overs/user-options-pop.jsx [hunk 2]
```diff
@@ -92,6 +92,7 @@ Are you sure you want to sign out?`)) {
         return;
       }
     }
+    togglePopover();
```
Comment (judeallred) [inline src/presenters/pop-overs/user-options-pop.jsx]: ?

Comment (GregWeil) [reply src/presenters/pop-overs/user-options-pop.jsx]: When you signed out and there was no currentUser the avatar menu would get removed from the dom. Now that there's always a currentUser we have to close it manually.

**Comment by ttseng** [comment]
cool this is a good idea.  is there anything in particular I should test out?

File: src/presenters/pop-overs/user-options-pop.jsx (hunk 2 above)
Comment (judeallred) [reply src/presenters/pop-overs/user-options-pop.jsx]: ah cool, thanks

**Pull request merged**